Felienne/hedy
# Use concrete command syntax in invalid type and invalid type combination error messages #1875

**Please fill out this template by replacing all content between < >**

< Use concrete command syntax in invalid type and invalid type combination error messages #1875 >

**Description**

< 
- Update the text of the invalid type and invalid type combination commands, so that they reflect better their dynamic content.
- Populate the error messages with the concrete keywords used in the program.
- Add type validation for repeat and for-in commands.
- Add tests.
 > Use present tense without a subject to describe your PR, for example: "Adds translations of levels 1 to 12 to Polish"

**Fixes < #1875  >**

< Always link the number of the issue or of the discussion that your pr concerns. >
Notable exception: adding content may be done without an accompanying issue 

**How to test**

<
Run Hedy locally, set the language to ES and execute the following code in level 3. Does the error message text give a clear indication of what is wrong and a suggestion of how to address it?
```
animales es perro, gato, canguro
imprimir animales
```
Also run the following in level 6. Is the message text clear when there is no keyword but a sign (+)?
```
name = 'test'
b = 5 + name
```
At level 12, check the error message for invalid type combination. Again, is the message clear?
```
test = 'bla'
print 2.5 + test
```

Now try using commands with multiple keywords, e.g. at random in level 3. Does the error correctly refer to the concrete syntax used? Is the error clear?
```
v is 1
print v at random
print v en aleatorio
print v at aleatorio
print v en random
```
 >

**Checklist**
Done? Check if you have it all in place using this list*
  

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
